internal/ingress/controller/template/template.go

```diff
@@ -1343,33 +1343,29 @@ func httpsListener(addresses []string, co string, tc config.TemplateConfig) []st
 	return out
 }
 
-func buildOpentracingForLocation(isOTEnabled bool, location *ingress.Location) string {
+func buildOpentracingForLocation(isOTEnabled bool, isOTTrustSet bool, location *ingress.Location) string {
 	isOTEnabledInLoc := location.Opentracing.Enabled
 	isOTSetInLoc := location.Opentracing.Set
 
 	if isOTEnabled {
 		if isOTSetInLoc && !isOTEnabledInLoc {
```

Comment:
I'm still trying to ramp up on this, but maybe... why we check isOTSetnLock and isOTEnabledInLoc, when isOTEnabledInLoc = false will always be false? And the default value is false anyway?

I would remove this and check only !isOTEnabledInLoc {return opentracing off; }

Reply:
The default for `isOTEnabledInLoc` is false, so this scenario means:
- OT is enabled globally
- There is an OT annotation present on the specific Ingress
- That annotation is set to false

The reason to check both is the (common) case where OT is enabled globally, and there is no annotation present on an Ingress. In that case the values will be: `{isOTEnabled: true, isOTSetInLoc: false, isOTEnabledInLoc: false}`, BUT it *should* be enabled in that location.